src/ansys/fluent/core/utils/data_transfer.py

```diff
@@ -110,15 +109,15 @@ def transfer_case(
             container_workdir = PurePosixPath(container_workdir)
     for idx in range(num_files_to_try):
         file_name_tmp = (file_name_stem or "temp_case_file") + "_" + str(idx)
-        folder = Path(tempfile.mkdtemp(prefix="temp_store-", dir=workdir))
-        file_name = folder / file_name_tmp
         if inside_container:
-            file_name_container = container_workdir / folder.name / file_name_tmp
-            network_logger.debug(f"file_name: {file_name}")
+            file_name_container = container_workdir / file_name_tmp
+            network_logger.debug(f"file_name: {file_name_tmp}")
```

Comment:
The usage of the temporary directory with random name was added by @prmukherj in #1053, need to check if just moving the files to the root directory won't have any side effects, considering the folder name was random but the file name is not. We should probably stick to putting these files inside isolated random directories if at all possible to avoid clashes and/or having to clean this up in between tests.

Also, I'm still a bit confused as to why the gRPC file transfer service can't just respect whatever path was specified here, and needs to have these files at the root level instead of allowing them to be inside folders.

Do we need to pass the `remote_file_name` for this part of the code so that the directory name is respected:
https://github.com/ansys/pyfluent/blob/0d99b218d67a12847bddafd2edaaabd3ee63331f/src/ansys/fluent/core/utils/file_transfer_service.py#L97-L99

Looking at this now, I am unsure if the blanket `basename()` method usage when `remote_file_name` is not specified was a good idea, why are we ignoring all of the file path that was specified, by default? Should we implement some logic that identifies the relative file path based on the current working directories instead?

Reply:
It should directly write file in `/mnt/pyfluent` not in any `/mnt/pyfluent/temp**/<file_name>`.

Due to this gRPC based server is not able to locate and open file because default path for all scripts is `/mnt/pyfluent`

Reply:
In `upload_file(file_name, remote_file_name)`,

`file_name` can be relative or absolute path.

`remote_file_name` can be only absolute file path or `os.path.basename(file_name)`

Reply:
Right, I quoted the wrong part of the code for this specific use-case, but the same comment still applies to both that situation and this one. This is the one being used here, still using `basename()`:
https://github.com/ansys/pyfluent/blob/65f9fd2eb244e097ba56786861067af77f0c51e6/src/ansys/fluent/core/utils/file_transfer_service.py#L140-L147

In both of these I don't think we should be using just `basename()` and losing information about the paths being used. 

I think we should instead be extracting and using the relative paths (relative to the current working directory both remotely and locally), as appropriate. That should allow us to work with subdirectories and not having to move all files to the root working directory like your changes propose here